src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeExperiment.groovy

```diff
@@ -14,239 +16,258 @@ import java.lang.reflect.Method
+   * @return OxfordNanoporeExperiment A new instance of a nanopore experiment.
+   */
+  static OxfordNanoporeExperiment create(Map nanoPoreSequencerOutput) {
+    final String sampleId = parseQbicIdFromRootFolder(nanoPoreSequencerOutput)
+    final List<OxfordNanoporeMeasurement> measurements = parseMeasurements(nanoPoreSequencerOutput)
+    return new OxfordNanoporeExperiment(sampleId, measurements)
+  }
+
+  /**
+   * Provides a list of measurements contained within the experiment folder.
+   * @return
+   */
+  List<OxfordNanoporeMeasurement> getMeasurements() {
+    return this.measurements
+  }
+
+  @Override
+  String getSampleCode() {
+    return this.sampleId
+  }
+
+  /**
+   * Helper method that parses the QBiC identifier from the root folder name
+   */
+  private static String parseQbicIdFromRootFolder(Map nanoPoreSequencerOutput) {
+    def name = Objects.requireNonNull(nanoPoreSequencerOutput.get("name"), "The root folder must contain a name property.")
+    final def ids = SampleCodeFunctions.findAllQbicSampleCodes(name as String)
+    if (ids.isEmpty()) {
+      throw new IllegalArgumentException("No QBiC sample identifier found!")
+    }
+    if (ids.size() > 1) {
+      throw new IllegalArgumentException("Name contained more than one valid sample id!")
     }
+    return ids.get(0)
+  }
 
-    @Override
-    String getSampleCode() {
-        return this.sampleId
+  /**
+   * Helper method that creates the measurements from the sequencer output
+   */
+  private static List<OxfordNanoporeMeasurement> parseMeasurements(Map nanoPoreSequencerOutput) {
+    final def measurements = []
+    Objects.requireNonNull(nanoPoreSequencerOutput.get("children"), "The root folder must contain at least one measurement folder.")
+    nanoPoreSequencerOutput.get("children").each { Map measurementItem ->
+      def name = measurementItem.get("name") as String
+      def relativePath = measurementItem.get("path") as String
+      def children = parseMeasurementItems(measurementItem.get("children") as List)
+      def metadata = measurementItem.get("metadata") as Map
+      measurements.add(new OxfordNanoporeMeasurement(name, relativePath, children, metadata))
     }
+    return measurements
+  }
 
-    /**
-     * Helper method that parses the QBiC identifier from the root folder name
-     */
-    private static String parseQbicIdFromRootFolder(Map nanoPoreSequencerOutput) {
-        def name = Objects.requireNonNull(nanoPoreSequencerOutput.get("name"), "The root folder must contain a name property.")
-        final def ids = SampleCodeFunctions.findAllQbicSampleCodes(name as String)
-        if (ids.isEmpty()) {
-            throw new IllegalArgumentException("No QBiC sample identifier found!")
+  /*
+   * Helper method that creates a list of mixed DataFolders and DataFiles instances
+   */
+
+  private static List<?> parseMeasurementItems(List<Map> items) {
+    final def children = []
+    items.each { item ->
```

Comment:
Why did you change from `IllegalArgumentException` to the differentiation by properties present?

Reply:
Basically to avoid unnecessary exception generation since we now know what defines a file/folder before iterating over all possible combinations and allow for unknown items in the map. 

**In more detail:** 
Since we now allow for optional files and folders, parseFile() and parseFolder() will not throw an exception if the item is not recognized. Instead it will create an OptionalFile with the content of the provided item. 
However this is not possible if the item is a folder, since a folder does not have a file_type, which is required for the optional file generation. This leads to an exception which would have to be caught and redirected again. 

Also IMO It's more explicit since we explicitly show what the defining properties are since we rely on them during the File/Folder generation process anyway.

Reply:
The original reasoning was that what makes a file a file and what makes a folder a folder is defined in the JSON schema. Thus only the method that uses the schema can determine whether we are indeed parsing a file or a folder. It first checks whether a file can be parsed, if not it tries to parse a folder. If neither works, then there should be an exception. The solution suggested in this PR does not consider that and instead hard-codes the properties.

Reply:
As we have this logic in multiple places, perhaps it would be an option to extract the parsing or to restructure the code slightly.
If this is more complex than suspected or if I missed some aspects of it, I will not block this PR as it works.

Reply:
I'd agree with you however the parseFile and parseFolder methods beforehand also expected specific properties to be there in the first place (namely "name", "path", and "children"). 
Additionally the parsing should not happen if the schema fails, therefore we can expect that these properties could be found during the validation process.

Reply:
I'd argue that down the line we should re-evaluate where the schema should be processed. 
IMO the strict seperation of schema and parsing does not make sense anymore and the schema should be automatically consulted if a NanoporeExperiment/NanoporeMeasurement is generated. 🤷‍♂️